db/python/layers/ourdna/dashboard.py

```diff
@@ -0,0 +1,140 @@
+from collections import defaultdict
+from datetime import datetime
+from typing import Any
+
+from db.python.connect import Connection
+from db.python.layers.base import BaseLayer
+from db.python.tables.sample import SampleFilter, SampleTable
+from models.models.sample import Sample, SampleInternal
+
+
+class OurDnaDashboardLayer(BaseLayer):
+    """Layer for analysis logic"""
+
+    def __init__(self, connection: Connection):
+        super().__init__(connection)
+
+        # TODO initialize other layers here as needed for aggregation
+        # self.at = AnalysisRunnerTable(connection)
+
+        self.sample_table = SampleTable(connection)
+
+    async def get_collection_to_process_end_time(self, sample: Sample) -> int | None:
```

Comment:
FWIW, if you don't expect this to be nullable, I'd probably not mark it as so. It can give a false illusional it's making a async check.

Also, definitely not expecting these to already be here, but these methods are very test-able. Also consider the cases where the values are not in the expected format.

Reply:
I think where `process_end_time` is not available yet because the sample isn't processed, or it is lost, we wouldn't be able to make this calculation. And so we have two cases here to be handled: not processed/lost OR processed. Thoughts?

Reply:
Yeah totally agree, just make sure it's handled as strptime will raise a ValueError.
(I meant marking this as async, not nullable, got my words all mixed up)

Reply:
Removed the async :) I think the handling should be working fine but will include this check in testing later.